src/SystemLog/Logger.cs

```diff
@@ -39,7 +40,10 @@ void ILogPlugin.Log(string source, LogLevel level, string message)
 
                 if (!string.IsNullOrEmpty(Settings.Default.Path))
                 {
-                    var path = Combine(Settings.Default.Path, source);
+                    StringBuilder sb = new StringBuilder(source);
+                    foreach (char c in GetInvalidFileNameChars())
+                        sb.Replace(c, '-');
```

Comment:
```suggestion
                        sb = sb.Replace(c, '-');
```

Reply:
It doesn't returns new `StringBuilder`. It returns the old one. There will be no new one be created.